pandas/core/arrays/categorical.py

```diff
@@ -1950,6 +1952,35 @@ def __contains__(self, key) -> bool:
     # ------------------------------------------------------------------
     # Rendering Methods
 
+    def _format_array(
+        self,
+        formatter: Callable | None,
+        float_format: FloatFormatType = None,
+        na_rep: str = "NaN",
+        digits: int = None,
+        space: str | int = None,
+        justify: str = "right",
+        decimal: str = ".",
+        leading_space: bool | None = True,
+        quoting: int | None = None,
+    ) -> list[str]:
+        from pandas.io.formats.format import format_array
+
+        array = self._internal_get_values()
```

Comment:
too bad the indirection can't be handled in the base class

Reply:
We could have a kind of `_values_for_format_array`, and the default would be `np.asarray(self)`. Then we could have`_format_array` `_values_for_format_array`.

I do think that giving the EA author control of turning the array into a List[str] is important though, so I wouldn't want just `_values_for_format_array`.

Reply:
sure i get it. though can we reuse a `_values_for....` anywhere?

Reply:
Right now we have `_values_for_factorize` and `_values_for_argsort`, neither of which would be what we need here I think. It would require a new method, which I'm happy to add if we think it's worthwhile.